float64 xp
float64 yp
bool blob 